pkg/segment/segment_test.go

```diff
@@ -241,6 +252,51 @@ func TestSetError(t *testing.T) {
 	}
 }
 
+func TestIsTelemetryEnabled(t *testing.T) {
+	tests := []struct {
+		errMesssage, envVar   string
+		want, preferenceValue bool
+	}{
+		{
+			want:            false,
+			errMesssage:     "Telemetry must be disabled.",
+			envVar:          "true",
+			preferenceValue: false,
+		},
+		{
+			want:            false,
+			errMesssage:     "Telemetry must be disabled.",
+			envVar:          "false",
+			preferenceValue: false,
+		},
+		{
+			want:            false,
+			errMesssage:     "Telemetry must be disabled.",
+			envVar:          "true",
+			preferenceValue: true,
+		},
+		{
+			want:            true,
+			errMesssage:     "Telemetry must be enabled.",
+			envVar:          "false",
+			preferenceValue: true,
+		},
```

Comment:
One other case would be with a invalid value for env var such that `strconv.ParseBool()` throws an error.

Reply:
I was planning on adding that, but then I figured we're anyway ignoring the error from `strconv.ParseBool()`, then why check for it. If there is an error, it will consider it False. If you still think it will be good to have, let me know.

Reply:
If someone modifies the code to
```
disableTelemetry, err:= strconv.ParseBool(os.Getenv(DisableTelemetryEnv))
	if err!=nil {
		return true
	}
```
In case of invalid value like `some-value`, `ParseBool` would throw an error and the function would return true but the current test cases won't detect it.